rest/client.go

```diff
@@ -241,74 +264,60 @@ func (c Client) postOrPut(
 
 	statusCode := resp.StatusCode()
 	if err == nil {
-		switch statusCode {
-		case http.StatusOK:
-			if resp.Header.ContentLength() > 0 {
-				return parseFn(resp)
-			}
-		case http.StatusCreated:
-			if resp.Header.ContentLength() > 0 {
-				return parseFn(resp)
-			}
-		case http.StatusAccepted:
-			return nil
-		default:
-			// if possible, still populate using response body if there is one
-			_ = parseFn(resp)
-			return NewHTTPError(statusCode, string(resp.Body()))
-		}
+		return handleResponse(statusCode, resp, parseFn)
 	}
+
 	return handleError(err)
 }
 
-func (c Client) get(
-	ctx context.Context,
-	uri string,
-	parseFn responseParseFn,
-	headers map[string]string,
-	query map[string]string) error {
-	req := fasthttp.AcquireRequest()
-	defer fasthttp.ReleaseRequest(req)
-	req.SetRequestURI(uri)
-
-	ctx, span := otel.Tracer(tracerName).Start(ctx, string(req.URI().Path()))
-	defer span.End()
-
-	for k, v := range query {
-		req.URI().QueryArgs().Add(k, v)
-	}
-	req.Header.SetMethod(fasthttp.MethodGet)
-	for k, v := range headers {
-		req.Header.Add(k, v)
+func handleResponse(statusCode int, resp *fasthttp.Response, parseFn responseParseFn) error {
+	switch statusCode {
+	case http.StatusOK:
+		if resp.Header.ContentLength() > 0 {
+			return parseFn(resp)
+		}
+	case http.StatusCreated:
+		if resp.Header.ContentLength() > 0 {
+			return parseFn(resp)
+		}
+	case http.StatusAccepted:
+		return nil
+	default:
+		// if possible, still populate using response body if there is one
+		_ = parseFn(resp)
+		return NewHTTPError(statusCode, string(resp.Body()))
 	}
-	addTraceHeaders(ctx, &req.Header)
-
-	resp := fasthttp.AcquireResponse()
-	defer fasthttp.ReleaseResponse(resp)
-
-	log.Ctx(ctx).Debug().Func(ops.LogHTTPRequest(req)).Msg("http client request")
+	return nil
+}
 
-	err := c.fastClient.DoTimeout(req, resp, c.config.RequestTimeout)
-	if err != nil {
-		log.Ctx(ctx).Error().Func(ops.LogHTTPResponse(req, resp, err)).Msg("http client response")
-	} else {
-		log.Ctx(ctx).Debug().Func(ops.LogHTTPResponse(req, resp, err)).Msg("http client response")
+// retry will run call() and check its returned error. Errors matching any of retriedErrors
+// are retried up to maxRetries amount of times.
+func retry(call func() error, maxRetries int, retriedErrors []error) (err error) {
+loop:
```

Comment:
first time I've used labels, horrible or wdyt?

Reply:
It is quite horrible but apparently idiomatic go, so who not 😄